src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs

```diff
@@ -309,23 +309,65 @@ private void DiscoverHubMethods()
         {
             var typeInfo = typeof(THub).GetTypeInfo();
 
-            foreach (var methodInfo in typeInfo.DeclaredMethods.Where(m => IsHubMethod(m)))
+            // used for checking against methods in base classes
+            var methods = new Dictionary<string, MethodInfo>();
```

Comment:
My main question would be if MVC blocks the use of the `new` keyword intentionally or if it is an unintentional omission. It does seem odd to me that we would half-support derived classes by supporting overriding but not shadowing.